### PR TITLE
CI fix for native aot tests

### LINKED ISSUE #69476: [NativeAOT] System.Reflection.Tests timeout/hang
```
----- start Wed May 18 01:17:17 UTC 2022 =============== To repro directly: =====================================================
pushd .
chmod +rwx System.Reflection.Tests ^&^& ./System.Reflection.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing 
popd
===========================================================================================================
/root/helix/work/workitem/e /root/helix/work/workitem/e
Running assembly:System.Reflection.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51

...
[EXECUTION TIMED OUT]
Exit Code:-3Executor timed out after 2700 seconds and was killed
```

### PR DESCRIPTION
**Pull request by LakshanF**
Fixes #69476

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
Oh interesting, the x64 checked NativeAOT failure is probably some CoreCLR bug in .NET 7 Preview 3 that we're using to run ilc.exe. I've not seen it before and we'll need to keep an eye for it. It's definitely unrelated to anything this pull request is touching. Filed #69593.

```
2022-05-19T20:19:42.4156624Z   Fatal error. Internal CLR error. (0x80131506)
2022-05-19T20:19:42.4158726Z      at Internal.TypeSystem.Ecma.EcmaModule+IEntityHandleObject.get_Handle()
2022-05-19T20:19:42.4171990Z      at Internal.TypeSystem.Ecma.EcmaModule+EcmaObjectLookupHashtable.CompareKeyToValue(System.Reflection.Metadata.EntityHandle, IEntityHandleObject)
2022-05-19T20:19:42.4207064Z      at Internal.TypeSystem.LockFreeReaderHashtable`2[[System.Reflection.Metadata.EntityHandle, System.Reflection.Metadata, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].TryGetValue(System.Reflection.Metadata.EntityHandle, System.__Canon ByRef)
2022-05-19T20:19:42.4226480Z      at ILCompiler.LazyGenericsSupport+GraphBuilder.WalkMethod(Internal.TypeSystem.Ecma.EcmaMethod)
2022-05-19T20:19:42.4234505Z      at ILCompiler.LazyGenericsSupport+GraphBuilder..ctor(Internal.TypeSystem.Ecma.EcmaModule)
2022-05-19T20:19:42.4240090Z      at ILCompiler.LazyGenericsSupport+CycleInfoHashtable.CreateValueFromKey(Internal.TypeSystem.Ecma.EcmaModule)
2022-05-19T20:19:42.4246260Z      at Internal.TypeSystem.LockFreeReaderHashtable`2[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].CreateValueAndEnsureValueIsInTable(System.__Canon)
2022-05-19T20:19:42.4249860Z      at ILCompiler.LazyGenericsSupport+GenericCycleDetector.DetectCycle(Internal.TypeSystem.TypeSystemEntity, Internal.TypeSystem.TypeSystemEntity)
2022-05-19T20:19:42.4255173Z      at Internal.JitInterface.CorInfoImpl.GetMethodEntrypoint(Internal.JitInterface.CORINFO_MODULE_STRUCT_*, Internal.TypeSystem.MethodDesc)
2022-05-19T20:19:42.4262677Z      at Internal.JitInterface.CorInfoImpl.getCallInfo(Internal.JitInterface.CORINFO_RESOLVED_TOKEN ByRef, Internal.JitInterface.CORINFO_RESOLVED_TOKEN*, Internal.JitInterface.CORINFO_METHOD_STRUCT_*, Internal.JitInterface.CORINFO_CALLINFO_FLAGS, Internal.JitInterface.CORINFO_CALL_INFO*)
2022-05-19T20:19:42.4276561Z      at Internal.JitInterface.CorInfoImpl._getCallInfo(IntPtr, IntPtr*, Internal.JitInterface.CORINFO_RESOLVED_TOKEN*, Internal.JitInterface.CORINFO_RESOLVED_TOKEN*, Internal.JitInterface.CORINFO_METHOD_STRUCT_*, Internal.JitInterface.CORINFO_CALLINFO_FLAGS, Internal.JitInterface.CORINFO_CALL_INFO*)
2022-05-19T20:19:42.4285443Z      at Internal.JitInterface.CorInfoImpl.JitCompileMethod(IntPtr ByRef, IntPtr, IntPtr, IntPtr, Internal.JitInterface.CORINFO_METHOD_INFO ByRef, UInt32, IntPtr ByRef, UInt32 ByRef)
2022-05-19T20:19:42.4288953Z      at Internal.JitInterface.CorInfoImpl.JitCompileMethod(IntPtr ByRef, IntPtr, IntPtr, IntPtr, Internal.JitInterface.CORINFO_METHOD_INFO ByRef, UInt32, IntPtr ByRef, UInt32 ByRef)
2022-05-19T20:19:42.4295063Z      at Internal.JitInterface.CorInfoImpl.CompileMethodInternal(ILCompiler.DependencyAnalysis.IMethodNode, Internal.IL.MethodIL)
2022-05-19T20:19:42.4302808Z      at Internal.JitInterface.CorInfoImpl.CompileMethod(ILCompiler.DependencyAnalysis.MethodCodeNode, Internal.IL.MethodIL)
2022-05-19T20:19:42.4308504Z      at ILCompiler.RyuJitCompilation.CompileSingleMethod(Internal.JitInterface.CorInfoImpl, ILCompiler.DependencyAnalysis.MethodCodeNode)
2022-05-19T20:19:42.4319081Z      at ILCompiler.RyuJitCompilation.CompileSingleMethod(ILCompiler.DependencyAnalysis.MethodCodeNode)
2022-05-19T20:19:42.4330825Z      at System.Threading.Tasks.Parallel+<>c__DisplayClass19_0`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<ForWorker>b__1(System.Threading.Tasks.RangeWorker ByRef, Int32, Boolean ByRef)
2022-05-19T20:19:47.6722523Z      at System.Threading.Tasks.TaskReplicator+Replica.Execute()
2022-05-19T20:19:47.6726731Z      at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
2022-05-19T20:19:47.6728763Z      at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread)
2022-05-19T20:19:47.6730129Z      at System.Threading.ThreadPoolWorkQueue.Dispatch()
2022-05-19T20:19:47.6731433Z      at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
```

**Comment by jkotas** [comment]
This is likely caused by https://github.com/dotnet/runtime/issues/68443 . We need to wait for the repo SDK to be updated to preview5. 

https://github.com/dotnet/runtime/issues/69454 and https://github.com/dotnet/runtime/issues/69192 are potentially related.

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -379,6 +379,7 @@
     <!-- Run only a small randomly chosen set of passing test suites -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj" />
     <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Collections\tests\System.Collections.Tests.csproj" />
+    <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Reflection\tests\System.Reflection.Tests.csproj" />
```
Comment (MichalStrehovsky) [inline src/libraries/tests.proj:382]: The hang when running the test seems to be Linux-specific, so following change should help us start getting coverage (and make sure we don't regress again) while the Linux hang is investigated:

```suggestion
    <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Reflection\tests\System.Reflection.Tests.csproj" Condition="'$(TargetOS)' == 'windows'" />
```

Comment (LakshanF) [reply src/libraries/tests.proj]: Thank!

**Review by MichalStrehovsky — APPROVED** [review]

**Comment by MichalStrehovsky** [comment]
Close/reopen to trigger CI. Infra was all red.

**Pull request merged**